katello/backup.rb

```diff
@@ -28,10 +28,18 @@ def initialize(foreman_proxy_content, foreman_proxy, program, accepted_scenarios
       setup_opt_parser
     end
 
+    # do not use run_cmd() inside this method, to avoid a recursive loop
+    # rather, use system() call provided by ruby
     def cleanup(exitstatus=-1)
       puts "Cleaning up backup folder and starting any stopped services... "
       FileUtils.cd("/")
       FileUtils.rm_rf @dir unless @options[:no_subdir]
+      if @options[:snapshot]
+        @databases.each do |database|
+          system("umount #{File.join(@mountdir, database)} >/dev/null")
+          system("lvremove -f #{get_snapshot_location(database)} >/dev/null")
```

Comment:
You should first get `get_snapshot_location(database)` and only execute `lvremove` if the get actually returned anything. Otherwise (when the snapshot in question does not exist), you end up with ugly errors:

    No command with matching syntax recognised.  Run 'lvremove --help' for more information.
    Correct command syntax is: